external/storm-hdfs/src/main/java/org/apache/storm/hdfs/avro/DefinedAvroSchemaRegistry.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.hdfs.avro;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaNormalization;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.Map;
+
+public class DefinedAvroSchemaRegistry implements AvroSchemaRegistry{
+
+    private final static String FP_ALGO = "SHA-256";
+    final BiMap<String, Schema> fingerprint2schemaMap;
+    final BiMap<Schema, String> schema2fingerprintMap;
+
+    DefinedAvroSchemaRegistry(final Map<String, Schema> definedSchemas) {
+        fingerprint2schemaMap = HashBiMap.create(definedSchemas);
+        schema2fingerprintMap = fingerprint2schemaMap.inverse();
+    }
+
+    DefinedAvroSchemaRegistry(List<Schema> schemaList) throws NoSuchAlgorithmException {
+        fingerprint2schemaMap = HashBiMap.create();
+        for (Schema schema : schemaList) {
+            String fingerPrint = new String(SchemaNormalization.parsingFingerprint(FP_ALGO, schema));
```

Comment:
I am not sure about constructing a string out of non-text bytes and then reading it back. What you write may be different from what you read back due to encoding/decoding difference. can you use a byte[] array as type instead of string for fingerprint? or if you really want to use string as the key, fingerprint can be base64 encoded and returned as string.